src/ui/Pager/Pager.ts

```diff
@@ -324,6 +324,15 @@ export class Pager extends Component {
     if (data.queryBuilder.numberOfResults == null) {
       data.queryBuilder.numberOfResults = eventArgs.count;
     }
+
+    // Prevents having a RequestedResultsMax error if we are requesting results over the maximumNumberOfResultsFromIndex option
```

Comment:
I think this comment can be removed if we extract the first two parts of the equation into a variable.
Also, because of the `Math.max` with `0`, the conditional is not needed.

 e.g.

```
const maxResultNumber = data.queryBuilder.firstResult + data.queryBuilder.numberOfResults;
const numOfExcessResults = Math.max(0, maxResultNumber - this.options.maximumNumberOfResultsFromIndex);

data.queryBuilder.numberOfResults -= numOfExcessResults;
```

Reply:
you're right

Reply:
Can `maxResultNumber - this.options.maximumNumberOfResultsFromIndex` ever be `<= 0`? If so, isn't this conditional gonna be interpreted as `false`?